src/app/dev/DevToys.Tools/Helpers/DateHelper.cs

```diff
@@ -0,0 +1,93 @@
+﻿using DevToys.Tools.Models;
+
+namespace DevToys.Tools.Helpers;
+
+internal static class DateHelper
+{
+    private static readonly DateTimeOffset minimumZoneOffsetTimestamp = DateTimeOffset.MinValue;
+
+    private static readonly DateTimeOffset maximumZoneOffsetTimestamp = DateTimeOffset.MaxValue;
+
+    internal static bool IsValidTimestamp(long value)
+    {
+        if (value < minimumZoneOffsetTimestamp.Millisecond || value > maximumZoneOffsetTimestamp.Millisecond)
+            return false;
+        return true;
+    }
+
+    internal static async Task<ToolResult<DateTimeOffset>> ConvertToDateTimeUtcAsync(
```

Comment:
isn't it a bit overkill to make sure to go off the UI thread in this method? From what I'm seeing, there's nothing in this code that should take more than 1 millisecond. `SwitchOffMainThreadAsync` is likely the slowest work being done here actually.